useful-scripts/misc-test-utils/src/services/promptService.ts

```diff
@@ -0,0 +1,223 @@
+import topicService from './topicService';
+import createPublication from '../modules/publication/createPublication';
+import createReleaseAndPublish from '../modules/publication/publishPublication';
+import uploadSingleSubject from '../modules/subject/uploadSubject';
+import createMethodology from '../modules/methodology/createMethodology';
+import addMethodlogyTextContentBlock from '../modules/methodology/addMethodlogyTextContentBlock';
+import addReleaseTextContentBlock from '../modules/release/addContentBlock';
+import logger from '../utils/logger';
+import releaseService from './releaseService';
+
+const promptService = {
+  createPublicationAndRelease: async () => {
+    await createPublicationAndRelease();
+  },
+
+  createRelease: async () => {
+    const publication = await prompt({
+      name: 'id',
+      type: 'input',
+      message: 'Enter publication id:',
+      prefix: '>',
+      validate: async input => {
+        if (!Guid.isGuid(input)) {
+          return 'Not a valid GUID';
+        }
+        return true;
+      },
+    });
+    await releaseService.createRelease(publication.id);
+  },
+
+  deleteThemeAndTopic: async () => {
+    const topic = await prompt({
+      name: 'id',
+      type: 'input',
+      message: 'Topic ID',
+      prefix: '>',
+      validate: async input => {
+        if (!Guid.isGuid(input)) {
+          return 'Not a valid GUID';
+        }
+        return true;
+      },
+    });
+    const theme = await prompt({
+      name: 'id',
+      type: 'input',
+      message: 'Theme ID',
+      prefix: '>',
+      validate: async input => {
+        if (!Guid.isGuid(input)) {
+          return 'Not a valid GUID';
+        }
+        return true;
+      },
+    });
+    await topicService.renameTopic(topic.id, theme.id);
+    await themeService.renameTheme(theme.id);
+    await themeService.deleteTheme(theme.id);
+  },
+
+  createPublication: async () => {
+    await createPublication();
+  },
+
+  createReleaseAndPublish: async () => {
+    await createReleaseAndPublish();
+  },
+
+  uploadSubject: async () => {
+    const release = await prompt({
+      name: 'id',
+      type: 'input',
+      message: 'Release ID from existing publication',
+      prefix: '>',
+      validate: async input => {
+        if (!Guid.isGuid(input)) {
+          return 'Not a valid GUID';
+        }
+        return true;
+      },
+    });
+
+    const numOfSubjects = await prompt({
+      name: 'number',
+      type: 'number',
+      message: 'How many subjects do you want to upload?',
+      prefix: '>',
+      default: 1,
+    });
+
+    const fast = await prompt({
+      name: 'shouldBeFast',
+      type: 'confirm',
+      message:
+        "would you like the uploadSubject function to exit as soon as a 'QUEUED' status is received? (this can be useful for uploading lots of subjects quickly)",
+    });
+
+    for (let i = 0; i < numOfSubjects.number; i += 1) {
+      // eslint-disable-next-line no-await-in-loop
```

Comment:
Be interesting to know what they suggest the better way of doing this would be - not for this PR though - just curious.

Reply:
I think the way to fix this is to use `Promise.all`. Doing this would be much faster but we want to basically perform this function sequentially since we can't try to rename multiple files in parallel with the current setup